handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -313,31 +333,58 @@ public void operationComplete(Future<SslContext> future) throws Exception {
     }
 
     /**
-     * The default implementation will simply call {@link AsyncMapping#map(Object, Promise)} but
-     * users can override this method to implement custom behavior.
+     * The default implementation will simply call {@link SslContextMapping#map(String, Promise)} but users can
```

Comment:
They're subject to type erasure but notice that the `IN` parameter of `SslContextMapping` is a concrete type with concrete argument types. There's no casting I could do to erase the String from the interface.

```java
interface SslContextMapping<T> extends AsyncMapping<String, T>
```